cmd/kubeadm/app/phases/certs/renewal/renewal.go

```diff
@@ -49,6 +53,77 @@ func RenewExistingCert(certsDir, baseName string, impl Interface) error {
 	return nil
 }
 
+// RenewEmbeddedClientCert loads a kubeconfig file, uses the renew interface to renew the client certificate
+// embedded in it, and then saves the resulting kubeconfig and key over the old one.
+func RenewEmbeddedClientCert(kubeConfigFileDir, kubeConfigFileName string, impl Interface) error {
+	kubeConfigFilePath := filepath.Join(kubeConfigFileDir, kubeConfigFileName)
+
+	// try to load the kubeconfig file
+	config, err := clientcmd.LoadFromFile(kubeConfigFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to load kubeconfig file %s", kubeConfigFilePath)
+	}
+
+	// get current context
+	if _, ok := config.Contexts[config.CurrentContext]; !ok {
```

Comment:
What's the behavior here is there are multiple contexts?

Reply:
Good catch! Fixed. Now the renew only updates the client certificate for the current context, but everything else is preserved